File: python/templative/lib/distribute/gameCrafter/advertisementCreator.py
from templative.lib.distribute.gameCrafter.util import httpOperations
from os import path
from templative.lib.distribute.gameCrafter.fileFolderManager import createFileInFolder
import pathlib

async def createAdvertisementFolder(gameCrafterSession, rootFolderId):
    folder = await httpOperations.postFolder(gameCrafterSession, "Advertisements", rootFolderId)
    return folder["id"]
    
async def createAdvertismentImageInFolder(gameCrafterSession, filepath, backupFilepath, folderId):
    filepathUsed = filepath
    if not path.exists(filepath):
        filepathUsed = backupFilepath
        print("!!! Advertising file %s doesn't exist, using default %s" % (filepath, backupFilepath))
    filenameWithoutExtension = pathlib.Path(filepathUsed).stem
    fileId = await createFileInFolder(gameCrafterSession, filenameWithoutExtension, filepathUsed, folderId)
    return fileId

async def createActionShot(gameCrafterSession, gameId, fileId):
    if fileId == "" or fileId == None:
        raise Exception("File id invalid.")
    await httpOperations.createActionShot(gameCrafterSession, gameId, fileId)

File: src/main/templative/lib/distribute/gameCrafter/advertisementCreator.js
const httpOperations = require('./util/httpOperations.js');
const { attemptCreateFileInFolder } = require('./fileFolderManager.js');
const path = require('path');
const fs = require('fs').promises;

async function createAdvertisementFolder(gameCrafterSession, rootFolderId) {
    const folder = await httpOperations.postFolder(gameCrafterSession, "Advertisements", rootFolderId);
    return folder.id;
}

async function createAdvertismentImageInFolder(gameCrafterSession, filepath, backupFilepath, folderId) {
    const filenameWithoutExtension = path.parse(filepath).name;
    const fileId = await attemptCreateFileInFolder(gameCrafterSession, filenameWithoutExtension, filepath, folderId);
    if (!fileId) {
        console.log(`!!! Advertising file ${filepath} doesn't exist, using default ${backupFilepath}`);
        const backupFilenameWithoutExtension = path.parse(backupFilepath).name;
        fileId = await attemptCreateFileInFolder(gameCrafterSession, backupFilenameWithoutExtension, backupFilepath, folderId);
    }
    return fileId;
}

async function createActionShot(gameCrafterSession, gameId, fileId) {
    if (fileId === "" || fileId === null) {
        throw new Error("File id invalid.");
    }
    await httpOperations.createActionShot(gameCrafterSession, gameId, fileId);
}

module.exports = { createAdvertisementFolder, createAdvertismentImageInFolder, createActionShot };
